.travis.yml

```diff
@@ -53,10 +53,12 @@ before_script:
 - node --version
 - rm -rf ./node_modules package-lock.json
 - travis_retry npm install
+- if [ "$PROM_METRICS_TEST" = "1" ]; then npm install prom-client@11.0.0; fi
```

Comment:
Yes via [`package.json#optionalDependencies`](https://docs.npmjs.com/files/package.json#optionaldependencies).

Nice to have the dependency more formally codified.

This will, however, attempt to install the Prom client every time the Jaeger node client is installed.

Reply:
Only if it's optional DEV dependency. We don't want it as optional runtime dependency.

Reply:
It would be installed as an optional runtime dependency.

Seems like the preferred scenario is to have the `PrometheusMetricsFactory` as a separate package which can have it's own dev dependencies. Jaeger node client would rely on `PrometheusMetricsFactory`, and when `PrometheusMetricsFactory` is installed the Prometheus client would not be installed.

Another benefit of this approach is the CI can run a matrix of tests not only on node versions but also various versions of the Prometheus client in addition to `@11.0.0`, like `@latest`. This is pretty attractive, IMO, since the Jaeger node client exerts no influence over the dependency.

Reply:
> Seems like the preferred scenario is to have the PrometheusMetricsFactory as a separate package which can have it's own dev dependencies.

Could you tell more about how PrometheusMetricsFactory is present as seperate package?  It will be great if any example is given.

Reply:
@eundoosong No problem. 

In [Facebook's jest](https://github.com/facebook/jest) test framework, the [`pretty-format`](https://github.com/facebook/jest/tree/master/packages/pretty-format) package functionality is implemented in a separate package. The package implements pretty-printing functionality for a number of data types including those within the [Immutable.js](https://facebook.github.io/immutable-js/) package, which is included as a [`devDependency`](https://github.com/facebook/jest/blob/master/packages/pretty-format/package.json#L18). Therefore, Immutable.js is [available in tests](https://github.com/facebook/jest/blob/master/packages/pretty-format/src/__tests__/immutable.test.js#L13) but is not installed when the pretty-print package is installed.

An added bonus is we might be able to install multiple versions of the Prometheus client via [yarn's alias feature](https://yarnpkg.com/en/docs/cli/add#toc-yarn-add-alias). This would allow us to maintain compatibility with every new version of the Prometheus client (including older versions, if we were so inclined).

This approach might work out well anytime we need to wrap packages that we don't want to include as dependencies.

Reply:
Would that require a separate repository?

Reply:
This can be done in a single repo using [lerna](https://lernajs.io/).

Reply:
as Joe and I discussed offline, we don't want to block this PR on the separate module approach

Reply:
Ok then, further improvement as tiffon said can be done as another issue?